security-model.md

```diff
@@ -0,0 +1,111 @@
+# Security Model
+
+This security model aims to clarify the project's current security position against industry 
+recommendations and general security best practices. The information provided here should make 
+it easier for security teams from highly regulated sectors to assess and use SPO for their workloads.
+
+The SPO project strives to maintain the [least privilege principle], by trying to run with 
+the bare minimum privileges for the least amount of time possible. This approach decreases
+the attack surface and the window of opportunity in which it can be abused.
+
+Note that this is an on-going effort, and may change over time as new features are developed.
+
+## Host Isolation (a.k.a. Pod Security / Security Context)
+
+Details the security requirements to run SPO from a host isolation perspective. 
+
+### Initialisation
+
+Seccomp leverages an init container to setup an initial symlink as `root`, so it can then operate
+ as a non-root user.
+
+During initialisation it also uses the capabilities:
+
+- `CHOWN`
+- `FOWNER`
+- `FSETID`
+- `DAC_OVERRIDE`
+
+
+### Running Mode
+
+The running permissions for the three core technologies supported: 
+
+|                                  | Seccomp     | SELinux     | AppArmor    |
+|----------------------------------|:-----------:|:-----------:|:-----------:|
+|               Requires root user | No          | Yes         | Yes         |
+|                 Requires HostPID | No          | No          | Yes         |
+|  Requires "privileged container" | No          | No          | Yes         |
+|     Requires SSH Access to nodes | No          | No          | No          |
+| Access to host's mount namespace | No          | Yes         | On Demand   |
+|                 AppArmor Profile | `default`   | `default`   | `default`   |
+|                     SELinux type | `spc_t`     | `spc_t`     | `spc_t`     |
+|                  Seccomp Profile | `unconfined`| `unconfined`| `unconfined`|
+
+
+#### Host Paths
+
+Throughout their operation they require read and write permissions into host paths:
+
+- `/var/lib/kubelet/seccomp`
+- `/etc/selinux.d`
+- `/etc/apparmor.d`
```

Comment:
There are a bunch of more host paths, should we cover them as well: https://cs.k8s.io/?q=HostPathVolumeSource&i=nope&files=&excludeFiles=vendor&repos=kubernetes-sigs/security-profiles-operator

Reply:
@saschagrunert nice! I did not know about `https://cs.k8s.io` :+1: 

Yes, I think we should break it all down and make it explicit what we need and to what level.
This will also help when we create our AppArmor profiles - which could actually become our official documentation on host-level disk permissions.